packages/@aws-cdk/aws-cognito-identitypool-alpha/lib/identitypool.ts

```diff
@@ -268,6 +267,94 @@ export interface IdentityPoolAuthenticationProviders {
   readonly customProvider?: string;
 }
 
+/**
+ * Map roles to users in the Identity Pool based on claims from the Identity Provider
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-cognito-identitypoolroleattachment.html
+ */
+export interface IdentityPoolRoleMapping {
+  /**
+   * The url of the Provider for which the role is mapped
+   */
+  readonly providerUrl: IdentityPoolProviderUrl;
+
+  /**
+   * The key used for the role mapping in the role mapping hash. Required if the providerUrl is a token.
+   * @default - The provided providerUrl
+   */
+  readonly mappingKey?: string;
+
+  /**
+   * If true then mapped roles must be passed through the cognito:roles or cognito:preferred_role claims from Identity Provider.
+   * @see https://docs.aws.amazon.com/cognito/latest/developerguide/role-based-access-control.html#using-tokens-to-assign-roles-to-users
+   *
+   * @default false
+   */
+  readonly useToken?: boolean;
+
+  /**
+   * Allow for role assumption when results of role mapping are ambiguous
+   * @default false - Ambiguous role resolutions will lead to requester being denied
+   */
+  readonly resolveAmbiguousRoles?: boolean;
+
+  /**
+   * The claim and value that must be matched in order to assume the role. Required if useToken is false
+   * @default - No role mapping rule
+   */
+  readonly rules?: RoleMappingRule[];
+}
+
+/**
+ * Types of matches allowed for role mapping
+ */
+export enum RoleMappingMatchType {
+  /**
+   * The claim from the token must equal the given value in order for a match
+   */
+  EQUALS = 'Equals',
+
+  /**
+   * The claim from the token must contain the given value in order for a match
+   */
+  CONTAINS = 'Contains',
+
+  /**
+   * The claim from the token must start with the given value in order for a match
+   */
+  STARTS_WITH = 'StartsWith',
+
+  /**
+   * The claim from the token must not equal the given value in order for a match
+   */
+  NOTEQUAL = 'NotEqual',
+}
+
+/**
+ * Represents an Identity Pool Role Attachment role mapping rule
+ */
+export interface RoleMappingRule {
+  /**
+   * The key sent in the token by the federated Identity Provider
+   */
+  readonly claim: string;
+
+  /**
+   * The role to be assumed when the claim value is matched
+   */
+  readonly mappedRole: IRole;
+
+  /**
+   * The value of the claim that must be matched
+   */
+  readonly claimValue: string;
+
+  /**
+   * How to match with the claim value
+   * @default RoleMappingMatchType.EQUALS
```

Comment:
nit: add line break between description and `@default`
```suggestion
   * How to match with the claim value
   *
   * @default RoleMappingMatchType.EQUALS
```